xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -1,16 +1,17 @@
 from typing import TypeVar
+from warnings import warn
```

Comment:
Interesting, this is the first time I see this, didn't even know about it.

IMO, we should make error reporting consistent throughout the repo. Maybe this is something sueful for @martin-luecke. AFAIU, you want this rewrite to go on, even if it hits this error. In this case, a warning should be printed. AFAIR, martin had the same problem and did it with a print. Should we change stuff to warn instead? What do you two think?

Reply:
Happy to discuss that yes! A happy flow in this case should just continue and hope to lower it later I guess. But then, I would like to have informative feedback in case the pipeline ends up in a broken state somehow. I have no clue how to do this, neither in MLIR or current xDSL.
I'd like to take care of it separately than this PR though, as we need this experimental work sooner rather than later. Same comment, maybe I can mark it as "to be taken care before getting out of experimental?"

Reply:
Ah, I did not know of this until now. 
I think this is much better than my usage of `print` statements, which can have effects on e.g. filecheck tests. 

+1 on having this consistent.

Reply:
Great, I love doing best practice by accident :slightly_smiling_face: 
Marking as resolved and opening an issue if not done yet!

Reply:
Actually, I did not get if that was done somewhere else in the main repo or not, I'm not sure how to phrase that issue. could someone else seeing what to improve exactly open it?